shell/apps/chrome-extension/chrome-data.js

```diff
@@ -0,0 +1,65 @@
+/*
+@license
+Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+
+import Xen from '../../components/xen/xen.js';
+
+import {deduplicate, filter, flatten} from './data-processing.js';
+
+class ChromeData extends Xen.Base {
+  static get observedAttributes() {
+    return ['arc'];
+  }
+
+  _update(props, state, lastProps, lastState) {
+    // if there isn't a listener attached for data, attach that
+    if (!state.listenerAttached) {
+      state.listenerAttached = window.addEventListener('message', event => {
+        if (event.source != window || event.data.method != 'injectArcsData' ||
+            !event.data || !event.data.entities) {
+          return;
+        }
+        ChromeData.log(
+            `received event ${event.data.method} from ${event.source}`,
+            event.data);
+
+        state.processedData = {
+          'entities': deduplicate(flatten(filter(event.data.entities)))
+        };
+        if (state.processedData.entities['text/x-arcs-manifest']) {
```

Comment:
Maybe a const for the mime type that can be used here and below?

Reply:
:+1: